src/subsplit_dag.hpp

```diff
@@ -60,6 +60,71 @@ class SubsplitDAG {
     SizeVector new_node_ids, new_edge_idxs, node_reindexer, edge_reindexer;
   };
 
+  // Nearest Neighbor Interchange Operations
+  class NNIOperation {
```

Comment:
If I understand correctly, this actually encodes two potential NNI operations: one for each clade of the child. Is that true?

Reply:
That was my original intent.  But as a DAG becomes more highly connected, for many of the pre-NNI parent/child pairs, one or both of the resulting NNIs are ineligible because they are already in the DAG.  So it seemed to me that the better course was just to store the explicit resulting parent/child pair of the NNI instead.

Reply:
Also, looking at this more, I'm going to suggest that we make a new source file. This file is BIG.

I generally like one class per file. If you want to include the SetOfNNIs I think that's OK. 

Should we update the contributing document?